### DISCUSSION TITLE
Indications that locals.scm entries are recognized/working? Could use some pointers.

### DISCUSSION THREAD

**Discussion by MostHated** [nvim-treesitter/nvim-treesitter#4662] [Q&A]
Hello all,
Forgive me if this is a silly question, but it has been a bit difficult to locate solid answers in relation to locals.scm usage and currently working functionality.

I have been working on a grammar and creating a locals.scm file, but I am having trouble knowing if it is actually working or doing what it is supposed to. 
I use tree-sitter playground, but am not sure if the information in there is supposed to be altered if a rule from the locals file is overriding a default rule/highlight?

Below is a small subset of what I was attempting to do. It has been difficult to know if I am even going about this the right way, as the information I have came across has been seemingly conflicting between the official tree-sitter site, the contributions.md file in this repo, and examples I came across out in the wild.

Sample code
```python

type Node1 object:
  a
  b

n1 = Node1{ a: 1, b: 2 }

t.eq(n1.a, 1)
```

highlights.scm

```scm
(typed_statement
  ((type_identifier) @reference.type
   (#is? @reference.type "type"))

```

locals.scm

```scm
(typed_statement
  ((type_identifier) @definition.type
   (#set! "definition.type.scope" "parent"))

```

Rule snippet:

```javascript
typed_statement: ($) => prec.left(PREC.object, seq(
  $.type,
  field("object_name", $.type_identifier),
  choice(
    $.identifier,
    alias($.field_expression, $.type_alias),
  ),
  choice(
    $.object_definition,
    $._newline,
  )
)),
```

I am pretty sure that what I have here is not correct, but I am hoping to get some sort of indication that I am at least headed in the right direction. 
While I don't necessarily need it to do anything differently at this point (though the first goal has been knowing that a variable is referencing the right object, and once I know things are working adding additional highlights for local variables, etc), I want to have everything setup properly so that I *can* utilize it in the near future. I just am not quite sure how to know if it *is* working properly.

If anyone could either give some quick feedback on if I should be able to see something different in tree-sitter playground or let me know if this is at least in the ball park, or even has a link to a known working example I could take a look at that is doing something similar, I would greatly appreciate it.

Thanks,
-MH

**Comment by theHamsta** [comment]
Hi! We've not added `#is?` because of the bad performance of the current implementation. Consider it an alpha feature. I do not remember whether `is?` supports subscope of reference.

**Comment by MostHated** [comment]
I appreciate the reply.

Ah, ok, that makes sense then I suppose. If that is the case, does detecting local variables in general not work as well? I was under the impression that either #is or #is-not was needed for that? 
(I want to say that I read that on the main tree-sitter site, or it was in a comment of a major languages tree-sitter files, or something along those lines)

Should the `(#set! "definition.type.scope" "parent")` part still work? If so, is what I tried to do above (minus the #is? line) the proper way to establish a reference to a @definition.whatever from the highlights.scm file (using @reference.whatever)?
